deepquest/utils/utils.py

```diff
@@ -0,0 +1,79 @@
+import codecs
+import logging
+
+from six import iteritems
+
+
+logging.basicConfig(level=logging.INFO,
+                    format='[%(asctime)s] %(message)s', datefmt='%d/%m/%Y %H:%M:%S')
+logger = logging.getLogger(__name__)
+
+
+def update_parameters(params, updates, restrict=False):
+    """
+    Updates the parameters from params with the ones specified in updates
+    :param params: Parameters dictionary to update
+    :param updates: Updater dictionary
+    :param restrict: If True, parameters from the original dict are not overwritten.
+    :return:
+    """
+    for new_param_key, new_param_value in iteritems(updates):
+        if restrict and params.get(new_param_key) is not None:
+            params[new_param_key] = new_param_value
+        else:
+            params[new_param_key] = new_param_value
+
+    return params
+
+
+def changes2dict(args):
+    import ast
+    if args.changes:
+        changes = {}
+        try:
+            for arg in args.changes:
+                try:
+                    k, v = arg.split('=')
+                except ValueError:
+                    print('Overwriting arguments must have the form key=value.\n This one is: %s' % str(changes))
+                    exit(1)
+                if '_' in v:
+                    changes[k] = v
+                else:
+                    try:
+                        changes[k] = ast.literal_eval(v)
+                    except ValueError:
+                        changes[k] = v
+        except ValueError:
+            print("Error processing arguments: {!r}".format(arg))
+            exit(2)
+        return changes
+    else:
+        return {}
+
+
+def setparameters(user_config_path, default_config_path='configs/default-config-BiRNN.yml'):
+    """
+    This function sets the overall parameters to used by the current running instance of dq.
+    """
+    try:
+        import yaml
+
+        with codecs.open('configs/default-config-BiRNN.yml', 'r', encoding='utf-8') as fh_default:
```

Comment:
If the path is set as a default argument `default_config_path`, then codecs.open() should take `default_config_path` as its argument. Although this will change in RSE-Sheffield/nmt-keras/pull/28 anyway

Reply:
exact, fixed with https://github.com/RSE-Sheffield/nmt-keras/pull/39/commits/e52a33939d68f88c1739abed77f69c558d7b1a4f.